superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterControl.tsx

```diff
@@ -165,28 +166,26 @@ const IconWrapper = styled.span`
   }
 `;
 
-interface DateFilterLabelProps {
+interface DateFilterControlProps {
   name: string;
   onChange: (timeRange: string) => void;
   value?: string;
   endpoints?: TimeRangeEndpoints;
-  datasource?: string;
+  datasource?: DatasourceMeta;
```

Comment:
Will update the shared control to pass in `datasource` object instead of datasource string from form_data.